packages/svelte-cloudinary/src/components/CldImage.svelte

```diff
@@ -1,90 +1,89 @@
+	/>
+	```
+-->
 
-	transformationPlugins.forEach(({ props = [] }) => {
-		props.forEach((prop) => {
-			if (CLD_OPTIONS.includes(prop)) {
-				throw new Error(`Option ${prop} already exists!`);
-			}
-			CLD_OPTIONS.push(prop);
-		});
-	});
+<script context="module" lang="ts">
+	import type { ImageProps } from '@unpic/svelte';
+	import type {
+		ConfigOptions,
+		ImageOptions,
+	} from '@cloudinary-util/url-loader';
 
-	$: imageProps = {
-		alt,
-		src,
-		width: typeof width === 'string' ? parseInt(width) : width,
-		height: typeof height === 'string' ? parseInt(height) : height
-	} as $$Props;
+	export type CldImageProps = ImageProps &
+		ImageOptions & {
+			/**
+			 * Overrides for the global Cloudinary config.
+			 * @see https://svelte.cloudinary.dev/config
+			 */
+			config?: ConfigOptions;
+		};
+</script>
 
-	$: if (imageProps) {
-		(Object.keys($$props) as Array<keyof $$Props>)
-			.filter((key) => !CLD_OPTIONS.includes(key))
-			.forEach((key) => {
-				//@ts-expect-error imageProps doesn't know the types of the keys
-				imageProps[key] = $$props[key];
-			});
-	}
+<script lang="ts">
+	import { pollForProcessingImage } from '@cloudinary-util/util';
+	import { createLoader } from '../helpers/loader';
+	import { Image } from '@unpic/svelte';
+
+	type $$Props = CldImageProps;
+
+	$: props = $$props as CldImageProps;
 
-	// Construct Cloudinary-specific props by looking for values for any of the supported prop keys
+	let key = 0;
+	async function handleError(event: Event) {
+		console.warn(
+			'[svelte-cloudinary]',
+			'image failed to load, polling for updates',
+		);
 
-	const cldOptions = {};
+		const src = (event.target as HTMLImageElement | null)?.src;
 
-	CLD_OPTIONS.forEach((key) => {
-		if ($$props[key]) {
-			// @ts-expect-error cldOptions doesn't know the types of the keys
-			cldOptions[key] = $$props[key] || undefined;
+		if (!src) {
+			console.warn('[svelte-cloudinary]', 'unable to find the image src');
+			return;
 		}
-	});
 
-	// Try to preserve the original transformations from the Cloudinary URL passed in
-	// to the component. This only works if the URL has a version number on it and otherwise
-	// will fail to load
-	$: if ($$props.preserveTransformations) {
-		try {
-			const transformations = getTransformations(imageProps.src).map((t) => t.join(','));
-			// @ts-expect-error rawTranformations key is not declared in cldOptions
-			cldOptions.rawTransformations = [
-				...transformations.flat(),
-				...($$props.rawTransformations || [])
-			];
-		} catch (e) {
-			console.warn(`Failed to preserve transformations: ${(e as Error).message}`);
+		const success = await pollForProcessingImage({ src });
+
+		console.warn(
+			'[svelte-cloudinary]',
+			success ? 'successfully loaded' : 'failed to load',
+			'image',
+		);
+
+		if (success) {
+			// Force image to update
+			key++;
 		}
 	}
 </script>
 
-{#if imageProps.src}
+{#key key}
 	<Image
-		{...imageProps}
+		transformer={createLoader(props)}
```

Comment:
for some reason this is showing up in the DOM
<img width="810" alt="image" src="https://github.com/user-attachments/assets/fb2f3f40-492d-4436-bd1b-c8abcb88afa6">

Reply:
so is CDN 
<img width="158" alt="image" src="https://github.com/user-attachments/assets/3e2e1174-4e04-40d0-87f9-69344a223238">

Reply:
also curious where g_center is coming from here
<img width="587" alt="image" src="https://github.com/user-attachments/assets/e3d6558d-5be8-40ce-bb2f-9f3d4925b6b0">

Reply:
I can't fix it rn without patching or making a pr to unpic

Reply:
might only happen on astro

Reply:
as long as it only happens in astro for now that works, i can try to find time to replicate it in a bare astro project and if so i'll report it